### PR TITLE
CloudWatch: dimension_values templating fix

### LINKED ISSUE #21215: 6.5.2 Filters not working for CloudWatch dimension_values
**What happened**:
After upgrading from 6.4.2 to 6.5.2, we noticed the filters parameter for `dimension_values()` seems to be ignored and instead returns all the results.

For example, this query will ignore the `FunctionName` filter and return every Lambda function in our account.
```
dimension_values(us-east-1,AWS/Lambda,Invocations,FunctionName, {"FunctionName": "jveldboom"})
```

**What you expected to happen**:
We expected the filter parameter to narrow down the search results.

**Environment**:
- Grafana version: v6.5.2
- Data source type & version: CloudWatch
- OS Grafana is installed on: Docker
- User OS & Browser: Chrome

### PR DESCRIPTION
**Pull request by sunker**
In Grafana 6.5, dimension values is an array instead of a string. This PR makes sure that the backend interpret the dimension value as an array, and also it makes sure that the wildcard (`*`) is never being passed to the templating function. 

fixes #21215

### CODE REVIEW

File: public/app/plugins/datasource/cloudwatch/components/QueryEditor.tsx [hunk 1]
```diff
@@ -161,7 +161,11 @@ export class QueryEditor extends PureComponent<Props, State> {
                   datasource.getDimensionKeys(query.namespace, query.region).then(this.appendTemplateVariables)
                 }
                 loadValues={newKey => {
-                  const { [newKey]: value, ...newDimensions } = query.dimensions;
+                  const { [newKey]: value, ...dim } = query.dimensions;
+                  const newDimensions = Object.entries(dim).reduce(
+                    (result, [key, value]) => (value === '*' ? result : { ...result, [key]: value }),
+                    {}
+                  );
```
Comment (torkelo) [inline public/app/plugins/datasource/cloudwatch/components/QueryEditor.tsx]: this part is a bit cryptic, maybe move loadValues into a seperate function, and add a comment about filtering out wildcard dimensions?

**Review by torkelo — APPROVED** [review]
Looks good, just one minor code readability improvement idea

File: public/app/plugins/datasource/cloudwatch/components/QueryEditor.tsx (hunk 1 above)
Comment (sunker) [reply public/app/plugins/datasource/cloudwatch/components/QueryEditor.tsx]: 👍

**Comment by marefr** [comment]
@sunker changed milestone to 6.5.3 since you've added the cherry pick needed label.

**Pull request merged**